golem-cli/tests/worker.rs

```diff
@@ -724,3 +725,38 @@ fn worker_list(
 
     Ok(())
 }
+
+fn worker_update(
+    (deps, name, cli): (
+        Arc<dyn TestDependencies + Send + Sync + 'static>,
+        String,
+        CliLive,
+    ),
+) -> Result<(), Failed> {
+    let cfg = &cli.config;
+
+    let component_id = make_component(deps, &format!("{name} worker_update"), &cli)?.component_id;
+    let worker_name = format!("{name}_worker_update");
+    let _: WorkerId = cli.run(&[
+        "worker",
+        "add",
+        &cfg.arg('w', "worker-name"),
+        &worker_name,
+        &cfg.arg('C', "component-id"),
+        &component_id,
+    ])?;
+    cli.run_unit(&[
+        "worker",
+        "update",
+        &cfg.arg('w', "worker-name"),
+        &worker_name,
+        &cfg.arg('C', "component-id"),
+        &component_id,
+        &cfg.arg('m', "mode"),
+        "auto",
+        &cfg.arg('t', "target-version"),
+        "1",
+    ])?;
+
```

Comment:
Please check the new version after update via `worker list` command.

Reply:
check added, thanks.

Reply:
There is a race condition here because the pending update will become a failed update (as you are trying to update to a non-existing version). Maybe the test will be flaky because of this.